packages/plugins/transaction-logger/src/index.js

```diff
@@ -47,16 +60,14 @@ export default class TransactionLogger {
     });
 
     this.writeLogFile = async.cargo((tasks, callback) => {
-      // TODO change this to only read once then use memory, because it slows things down a lot to read on each TX
-      const data = this._readLogs();
-
+      let appendThis = '';
       tasks.forEach(task => {
-        data[new Date().getTime()] = task;
+        appendThis += `"${new Date().getTime()}":${JSON.stringify(task, getCircularReplacer())},\n`;
```

Comment:
Why not use `util.inspect` instead? It handles circular structures and also has the added benefit of formatting with whitespacing instead of lumping it in to one string.

Reply:
I tried `util.isnpect`, but it is only useful to print. To store in a file, it has no use, because it doesn't output actual JSON. So when trying to read it, it was not able to parse it